app/controllers/ClaimPeriodQuestionController.scala

```diff
@@ -39,47 +36,41 @@ class ClaimPeriodQuestionController @Inject()(
   override val messagesApi: MessagesApi,
   sessionRepository: SessionRepository,
   val navigator: Navigator,
+  feature: FeatureFlagActionProvider,
   identify: IdentifierAction,
-  config: FrontendAppConfig,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
   formProvider: ClaimPeriodQuestionFormProvider,
   val controllerComponents: MessagesControllerComponents,
   view: ClaimPeriodQuestionView,
 )(implicit ec: ExecutionContext, errorHandler: ErrorHandler)
-    extends BaseController with ClaimPeriodQuestionRequestHandler {
+    extends BaseController {
 
   val form = formProvider()
 
-  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
-      val filledForm: Form[ClaimPeriodQuestion] =
-        request.userAnswers.get(ClaimPeriodQuestionPage).fold(form)(form.fill)
+  def onPageLoad(): Action[AnyContent] = (identify andThen feature(FastTrackJourneyFlag) andThen getData andThen requireData).async {
+    implicit request =>
+      getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
+        val filledForm: Form[ClaimPeriodQuestion] =
+          request.userAnswers.get(ClaimPeriodQuestionPage).fold(form)(form.fill)
 
-      onLoadResult(claimStart, claimEnd, filledForm)
-    }
+        Future.successful(Ok(view(filledForm, claimStart, claimEnd)))
+      }
   }
 
-  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
-      form
-        .bindFromRequest()
-        .fold(
-          formWithErrors => Future.successful(BadRequest(view(formWithErrors, claimStart, claimEnd))),
-          value => onSubmitResult(request, value)
-        )
-    }
+  def onSubmit(): Action[AnyContent] = (identify andThen feature(FastTrackJourneyFlag) andThen getData andThen requireData).async {
+    implicit request =>
+      getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
+        form
+          .bindFromRequest()
+          .fold(
+            formWithErrors => Future.successful(BadRequest(view(formWithErrors, claimStart, claimEnd))),
+            value =>
+              for {
```

Comment:
I don't agree on removing the handler - you have now added extra logic into the controller, made def bigger by removing the smaller defs and you also broke the pattern of delegating to a Request handler - I thought we have discussed in the PR for 150

Reply:
why do we need the handlers at all for the controllers ?